docs/specs/build/manifest.yml

```diff
@@ -14,16 +14,172 @@ outputs:
       "files":
       [
         {
-          "url": "/docs/a",
-          "path": "docs/a.json"
+          "siteUrl": "/docs/a",
+          "outputPath": "docs/a.json"
         },
         {
-          "url": "/docs/folder/b",
-          "path": "docs/folder/b.json"
+          "siteUrl": "/docs/folder/b",
+          "outputPath": "docs/folder/b.json"
         },
         {
-          "url": "/docs/TOC.json",
-          "path": "docs/TOC.json"
+          "siteUrl": "/docs/TOC.json",
+          "outputPath": "docs/TOC.json"
+        }
+      ]
+    }
+---
+# `build.manifest` contains Toc dependencies
+inputs:
+  docfx.yml:
+  docs/a.md:
+  docs/TOC.md: |
+    # [file reference](a.md)
+outputs:
+  docs/a.json:
+  docs/TOC.json:
+  build.manifest: |
+    {
+      "dependencies":
+      [
+        {
+          "source": "docs/TOC.md",
+          "dependencies": 
+          [
+            {
+              "source": "docs/a.md",
+              "type": "tocLink"
+            }
+          ]
+        }
+      ]
+    }
+---
+# `build.manifest` contains Inclusion dependencies
+inputs:
+  docfx.yml:
+  docs/a.md: a [!INCLUDE[](b.md)]
+  docs/b.md:
+outputs:
+  docs/a.json:
+  docs/b.json:
+  build.manifest: |
+    {
+      "dependencies":
+      [
+        {
+          "source": "docs/a.md",
+          "dependencies": 
+          [
+            {
+              "source": "docs/b.md",
+              "type": "inclusion"
+            }
+          ]
+        }
+      ]
+    }
+---
+# `build.manifest` contains File(Link) dependencies
+inputs:
+  docfx.yml:
+  docs/a.md: Link to [b](b.md)
+  docs/b.md:
+outputs:
+  docs/a.json:
+  docs/b.json:
+  build.manifest: |
+    {
+      "dependencies":
+      [
+        {
+          "source": "docs/a.md",
+          "dependencies": 
+          [
+            {
+              "source": "docs/b.md",
+              "type": "link"
+            }
+          ]
+        }
+      ]
+    }
+---
+# `build.manifest` contains nested(token/codesnippet) dependencies
+inputs:
+  docfx.yml: |
+    content: docs/a.md
+  docs/a.md: a [!INCLUDE[](b.md)
+  docs/b.md: |
+    Link to [c](c.md)
+    b [!INCLUDE[](c.md)
+  docs/c.md:
+outputs:
+  docs/a.json:
+  build.manifest: |
+    {
+      "dependencies":
+      [
```

Comment:
>[ [](start = 6, length = 1)

is this order deterministic?

Reply:
yes, order by source

---
In reply to: [193960577](https://github.com/dotnet/docfx/pull/2859#discussion_r193960577) [](ancestors = 193960577)